scheduler/test/cook/test/simulator.clj

```diff
@@ -325,12 +340,42 @@
                                                  :straggler-trigger-chan (async/chan)
                                                  :lingering-task-trigger-chan (async/chan)
                                                  :cancelled-task-trigger-chan (async/chan)}})
-        {:keys [time-ms-between-optimizer-calls time-ms-between-rebalancing]} config]
+        {:keys [time-ms-between-incremental-output time-ms-between-optimizer-calls time-ms-between-rebalancing]} config]
+    ;; launch a thread to perform incremental writes of completed jobs
+    (when temp-out-trace-file
+      (let [last-start-time-atom (atom simulation-time)
+            write-thread (Thread.
+                           ^Runnable
+                           (fn write-completed-tasks []
```

Comment:
It might be nice to extract this to a `defn`

Reply:
Extracted helper function `write-completed-tasks`. I did not extract the whole function as I did not want to expose the `last-start-time-atom` atom.